src/Raven.Client/Documents/Session/Tokens/CounterIncludesToken.cs

```diff
@@ -0,0 +1,43 @@
+﻿using System.Collections.Generic;
+using System.Text;
+using Raven.Client.Extensions;
+
+namespace Raven.Client.Documents.Session.Tokens
+{
+    public class CounterIncludesToken : QueryToken
+    {
+        private readonly Parameters _queryParameters;
+        private readonly HashSet<string> _counters;
+
+        private CounterIncludesToken(HashSet<string> countres, Parameters queryParameters)
+        {
+            _counters = countres;
+            _queryParameters = queryParameters;
+        }
+
+        public static CounterIncludesToken Create(HashSet<string> countres, Parameters queryParameters)
```

Comment:
pass only queryParameter name. Check how we do it in other places.

e.g.
```
var whereToken = WhereToken.Create(WhereOperator.AllIn, fieldName, AddQueryParameter(array));
```

Reply:
actually there is a reason why i delayed adding the parameter to the point when we actually write the RQL.
I did like you suggested at first, the problem with that is that these kinds of queries:
```
session.Query<Order>()
	.Include(i => i
		.IncludeCounter("Downloads")
		.IncludeDocuments(x => x.Company))
	.Where(x => x.ShipTo.Country == "Brazil");
```
will generate :
```
from Orders where ShipTo.Country = $p1 
include Company,counters($p0)
```
because `Include` is computed before `Where`, but `where` comes first in RQL.

Reply:
It is not a problem that the order of parameters does not match.

Reply:
Yes, that isn't an issue.

Reply:
fixed